### ISSUE TITLE
Timeshape with oceans

### ISSUE THREAD

**Issue by c-mckenna** [RomanIakovlev/timeshape#4]
The 2018d release of timezone-boundary-builder has a version with oceans (timezones-with-oceans.geojson.zip)

Any plans to publish a version of Timeshape that includes oceans?

**Comment by RomanIakovlev** [comment]
There’s no such plan right now, but I think it could be done. A couple of questions must be answered first:

1. Are there corresponding `ZoneId` instances in `java.time`?
2. How should such artifact be published: a version of `net.iakovlev.timeshape` or a different artifact? If it should be a version, how to name it?

I need to think about this before promising anything. Suggestions are welcome.

**Comment by RomanIakovlev** [comment]
Ok, I've checked the version with oceans, and it seems everything works, with the time zones in oceans looking like `Etc/GMT+3`. Now the second question remain, how to publish such artifact. Potentially, in addition to the options mentioned in previous comment, there's a 3rd one: always publish with oceans, and don't publish without oceans. 

The cost of including the ocean's time zones is 1.5 MB (or roughly 9%) increase in artifact size and 4 MB (or roughly 2%) increase in heap usage at runtime. This cost looks acceptable to me, and it would be the simplest option to go forward, so maybe I'll do just that. However, I've put some efforts into driving the artifact size and heap usage down, because there were users for whom the high heap memory usage was a show-stopper, and therefore I'd wait a bit and see if someone would respond here with an objection.

To summarize, @c-mckenna could you please ping me here in a week or so, and if no one will object before that, I'll just publish the `2018d.4` with the oceans included.

**Comment by c-mckenna** [comment]
Thanks @RomanIakovlev 

I've got a couple of other ideas for packaging the ocean data separately, although the effort involved likely outweighs the relatively small increase in artifact size and heap usage to the existing artifact.

I'll check back in here next week.

**Comment by c-mckenna** [comment]
Hi @RomanIakovlev, just touching base to see if you're happy to proceed with publishing the ocean version of Timeshape?

**Comment by RomanIakovlev** [comment]
Hi @c-mckenna, yes, I think I'll proceed with the new release soon. Can't give you any fixed date now, but will try to not drag this for too long. Cheers!

**Comment by RomanIakovlev** [comment]
I've released version 2018d.4, which includes the oceans as well. Will take a couple of hours to reach Maven Central. Please give it a try and feel free to close the issue if the result works for you.

**Comment by RomanIakovlev** [comment]
Closed this by mistake, reopening.

**Comment by c-mckenna** [comment]
Thanks @RomanIakovlev, sorry for the delay in trying out the new version.

It looks like the ocean timezones are inverted, for example (lat, lon):
-45.97, 147.25 - Expected: Etc/GMT+10, Actual: Etc/GMT-10
51.33, -144.91 - Expected: Etc/GMT-10, Actual: Etc/GMT+10

**Comment by RomanIakovlev** [comment]
Could you please check if you're not mixing the order of latitude and longitude in the `net.iakovlev.timeshape.TimeZoneEngine.query(double latitude, double longitude)` method? Please keep in mind that some software, like QGIS, shows coordinates in different order: first longitude, then latitude, as in cartesian (X, Y) coordinates. I'll also double-check on my side.

**Comment by RomanIakovlev** [comment]
I've looked at the data, and I'm not really sure what's going on there. On one hand, the output of Timeshape corresponds to the contents of the source data (https://github.com/evansiroky/timezone-boundary-builder/releases). On the other hand, you're right that the Etc/GMT zones look out of place, and should have "+" rather than "-" to the east from Greenwich.

Attaching the picture of how it looks in QGIS with the original geojson data. If that's indeed wrong, we need to report this to the source data project. @c-mckenna what do you think?

![timezones](https://user-images.githubusercontent.com/2363458/42987246-d6787fa4-8bf8-11e8-8f3a-34424e30480b.jpg)

**Comment by RomanIakovlev** [comment]
Actually, as you can read in the discussion in the issue referenced above, current behavior is actually correct. This comment in the time zone database source code should help: https://github.com/eggert/tz/blob/master/etcetera#L38-L53.
